File: easy/9_palindrome_number.rb
# 9. Palindrome Number
#
# Given an integer x, return true if x is palindrome integer.
#
# An integer is a palindrome when it reads the same backward as forward. For
# example, 121 is palindrome while 123 is not.

def is_palindrome(x)
  x == x.to_s.reverse.to_i
end

p is_palindrome(121)   # true
p is_palindrome(-121)  # false
p is_palindrome(10)    # false
p is_palindrome(-101)  # false

File: go/easy/9_palindrome_number.go
// 9. Palindrome Number
//
// Given an integer x, return true if x is palindrome integer.
//
// An integer is a palindrome when it reads the same backward as forward. For
// example, 121 is palindrome while 123 is not.
package main

func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}

	if x < 10 {
		return true
	}

	original := x
	y := 0

	for x != 0 {
		y *= 10
		y += x % 10
		x = x / 10
	}

	if original-y == 0 {
		return true
	}

	return false
}
